spec.bs

```diff
@@ -588,48 +590,77 @@ members:
         automatically.
 </dl>
 
-### Steps to check if a payment can be made ### {#sctn-steps-to-check-if-a-payment-can-be-made}
+### Payment Method additional data type ### {#sctn-payment-method-additional-data-type}
 
-The [=steps to check if a payment can be made=] for this payment method, for an
+The [=payment method additional data type=] for this payment method is
+{{SecurePaymentConfirmationRequest}}.
+
+### Steps to validate payment method data ### {#sctn-steps-to-validate-payment-method-data}
+
+The [=steps to validate payment method data=] for this payment method, for an
 input {{SecurePaymentConfirmationRequest}} |data|, are:
 
 <wpt>
   constructor.https.html
-  authentication-invalid-icon.https.html
-  authentication-icon-data-url.https.html
-  authentication-rejected.https.html
 </wpt>
 
-
 1. If |data|["{{SecurePaymentConfirmationRequest/credentialIds}}"] is empty,
-    return `false`.
+    throw a {{RangeError}}.
+
+1. If |data|["{{SecurePaymentConfirmationRequest/challenge}}"] is null or
+    empty, throw a {{TypeError}}.
 
 1. If |data|["{{SecurePaymentConfirmationRequest/rpId}}"] is not a
-    [=valid domain=], return `false`.
+    [=valid domain=], throw a {{TypeError}}.
 
 1. If both |data|["{{SecurePaymentConfirmationRequest/payeeName}}"] and
     |data|["{{SecurePaymentConfirmationRequest/payeeOrigin}}"] are omitted,
-    return `false`.
+    throw a {{TypeError}}.
 
 1. If either of |data|["{{SecurePaymentConfirmationRequest/payeeName}}"] or
     |data|["{{SecurePaymentConfirmationRequest/payeeOrigin}}"] is present and
-    empty, return `false`.
+    empty, throw a {{TypeError}}.
 
 1. If |data|["{{SecurePaymentConfirmationRequest/payeeOrigin}}"] is present:
 
     1. Let |parsedURL| be the result of running the [=URL parser=] on
         |data|["{{SecurePaymentConfirmationRequest/payeeOrigin}}"].
 
-    1. If |parsedURL| is failure, then return `false`.
+    1. If |parsedURL| is failure, then throw a {{TypeError}}.
 
-    1. If |parsedURL|'s [=url/scheme=] is not "`https`", then return `false`.
+    1. If |parsedURL|'s [=url/scheme=] is not "`https`", then throw a {{TypeError}}.
+
+1. If |data|["{{SecurePaymentConfirmationRequest/instrument}}"]["{{PaymentCredentialInstrument/displayName}}"]
+    is empty, throw a {{TypeError}}.
+
+### Steps to check if a payment can be made ### {#sctn-steps-to-check-if-a-payment-can-be-made}
+
+The [=steps to check if a payment can be made=] for this payment method, for an
+input {{SecurePaymentConfirmationRequest}} |data|, are:
+
+<wpt>
+  constructor.https.html
+  authentication-invalid-icon.https.html
+  authentication-icon-data-url.https.html
+  authentication-rejected.https.html
+</wpt>
+
+1. If |data|["{{SecurePaymentConfirmationRequest/payeeOrigin}}"] is present:
```

Comment:
Comparing to the [Chrome code](https://source.chromium.org/chromium/chromium/src/+/main:components/payments/content/secure_payment_confirmation_app_factory.cc;l=44-89;drc=cfd951a304bb7a1e6424c614c094f5bc581ca8b7) that determines whether `canMakePayment()` will return true:

1. Credential IDs list: non-empty. (Fails only on compromised renderer.) ✅
2. Each credential ID: non-empty. ([The `PaymentRequest()` constructor](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/modules/payments/secure_payment_confirmation_helper.cc;l=40-101;drc=cfd951a304bb7a1e6424c614c094f5bc581ca8b7) should be checking this, too.) 🔥
3. Timeout: under 1 hour. (Fails only on compromised renderer.) ✅
4. Challenge: non-empty. ([The `PaymentRequest()` constructor](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/modules/payments/secure_payment_confirmation_helper.cc;l=40-101;drc=cfd951a304bb7a1e6424c614c094f5bc581ca8b7) should be checking this, too.) 🔥
5. Instrument display name: non-empty. (Fails only on compromised renderer.) ✅
6. Instrument icon: valid URL. (Fails only on compromised renderer.) ✅
7. Download the icon.
8. Check for instrument IDs on device.

This spec here has:

1. Payee origin: valid https URL. (Chrome should check that either payeer origin is a valid https URL or payee name is non empty in [here](https://source.chromium.org/chromium/chromium/src/+/main:components/payments/content/secure_payment_confirmation_app_factory.cc;l=44-89;drc=cfd951a304bb7a1e6424c614c094f5bc581ca8b7).) 🔥
2. Download the icon. ✅
3. Check for instrument IDs on device. ✅